src/main/java/com/elixr/poc/service/UserService.java

```diff
@@ -21,53 +22,76 @@ public class UserService {
 
     public UserService(UserRepository userRepository) {
         this.userRepository = userRepository;
-
     }
 
     /**
      * Deleting the user by the userId.
-     * throwing a NoRecordFoundException to handel if the UserId is not present.
+     * Throwing a NoRecordFoundException to handel if the UserId is not present.
+     * @param userId
+     * @return
+     * @throws IdNotFoundException
      */
-    public boolean deleteUserDetails(UUID userId) throws NoRecordFoundException {
+    public boolean deleteUserDetails(UUID userId) throws IdNotFoundException {
         boolean success = false;
         boolean userRecordExists = userRepository.existsById(userId);
         if (userRecordExists) {
             userRepository.deleteById(userId);
             success = true;
         } else {
-            throw new NoRecordFoundException(ApplicationConstants.ID_MISMATCH);
+            throw new IdNotFoundException(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_ID_DOES_NOT_EXISTS.getKey()));
         }
-        return true;
+        return success;
     }
 
     /**
      * Creating a valid user
      * @param user
      * @return
      */
-    public ErrorResponse createValidUser(User user) {
+    public UserResponse createValidUser(User user) {
         saveDataToDatabase(user);
-        return UserResponse.builder().success(true).id(user.getId()).userName(user.getUserName()).firstName(user.getFirstName()).lastName(user.getLastName()).build();
+        return UserResponse.builder().success(true).id(user.getId()).userName(user.getUserName()).firstName(user
+                .getFirstName()).lastName(user.getLastName()).build();
     }
 
     /**
      * Calling the repository to store data
      * @param user
      * @return
      */
-    private void saveDataToDatabase(User user) {
+    private User saveDataToDatabase(User user) {
 
         if (user.getId() == null || user.getId().toString().isEmpty()) {
             user.setId(UUID.randomUUID());
         }
-        this.userRepository.save(user);
+        user = this.userRepository.save(user);
+        return user;
+    }
+
+    /**
+     * Finding User by userId and returning the user.
+     * @param userId
+     * @return
+     * @throws IdNotFoundException
+     */
+    public User getUserByUserId(UUID userId) throws IdNotFoundException {
+        Optional<User> user = userRepository.findById(userId);
+        return user.orElseThrow(() -> new IdNotFoundException(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_ID_DOES_NOT_EXISTS.getKey())));
```

Comment:
Please explain how does this line work

Reply:
Searching the user in db by its userId, if user is present with the particular userId then returning the user, if user is not found then throwing an exception IdNotFoundException